lib/MusicBrainz/Server/Controller/Role/EditRelationships.pm

```diff
@@ -191,8 +191,8 @@ role {
 
                 $args{entity0} = $field->{forward} ? $source : $target;
                 $args{entity1} = $field->{forward} ? $target : $source;
-                $args{entity0_credit} = $field->{entity0_credit};
-                $args{entity1_credit} = $field->{entity1_credit};
+                $args{entity0_credit} = $field->{entity0_credit} if exists $field->{entity0_credit};
```

Comment:
Right, FormHandler will not include keys for fields that weren't submitted, which is why `exists` is used there.

The code below is slightly different, 'cause there are multiple places in the code that call `Data::Relationship::update`. I haven't checked thoroughly, but I'm sure some `undef` credits will end up there, so I used `defined` because `undef` isn't valid either way.